pyracf/common/security_admin.py

```diff
@@ -286,7 +288,7 @@ def _add_traits_directly_to_request_xml_with_no_segments(
                     None, traits, self._trait_map, alter=alter
                 )
         # Clear state for subsequent requests
-        self._clear_state()
+        self.__clear_state()
```

Comment:
I don't think this is necessary since state is cleared at the end of `_make_request`.

Reply:
It still was necessary, since there was no state clearing for `__generate_requests_only`, which the unit tests originally ran on. I changed this code

```python
        if self.__generate_requests_only:
            return self.__logger.redact_request_xml(
                security_request.dump_request_xml(encoding="utf-8"),
                secret_traits=self.__secret_traits,
            )
```
to the following.

```python
        if self.__generate_requests_only:
            request = self.__logger.redact_request_xml(
                security_request.dump_request_xml(encoding="utf-8"),
                secret_traits=self.__secret_traits,
            )
            self.__clear_state(clear_preserved_segment_traits=True)
            del security_request
            return request
```

This should happen anyway since we DO still want to clean up the inputs when we are generating a request only.